cleverhans/utils_tf.py

```diff
@@ -200,10 +199,12 @@ def model_eval(sess, x, y, model, X_test, Y_test, args=None):
 
             # The last batch may be smaller than all others, so we need to
             # account for variable batch size here
-            accuracy += cur_batch_size * acc_value.eval(
+            cur_acc = acc_value.eval(
                 feed_dict={x: X_test[start:end],
                            y: Y_test[start:end],
                            keras.backend.learning_phase(): 0})
+
+            accuracy += (cur_batch_size * cur_acc).mean()
```

Comment:
I think it would make more sense to put a `tf.reduce_mean` at line 179. That way the shape of `acc_value` is the same regardless of keras version, so there is less variability in the behavior of our `cleverhans` code. The way it is right now, we need to keep track of two different possible shapes for two different possible `keras` versions, so it's a little more confusing.

I'd also suggest putting a comment at line 179 explaining why the `reduce_mean` is needed for new keras but not needed for old keras.